src/plone/api/relation.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+"""Module that provides functionality for relations.
+
+Heavily inspired by collective.relationhelpers.
+"""
+from AccessControl.SecurityManagement import getSecurityManager
+from collections import defaultdict
+from plone.api.exc import InvalidParameterError
+from plone.api.validation import at_least_one_of
+from plone.api.validation import required_parameters
+from plone.dexterity.interfaces import IDexterityContent
+from plone.dexterity.interfaces import IDexterityFTI
+from plone.dexterity.utils import iterSchemataForType
+from z3c.relationfield import event
+from z3c.relationfield import RelationValue
+from z3c.relationfield.schema import Relation
+from z3c.relationfield.schema import RelationChoice
+from z3c.relationfield.schema import RelationList
+from zc.relation.interfaces import ICatalog
+from zope.component import getUtility
+from zope.component import queryUtility
+from zope.intid.interfaces import IIntIds
+from zope.lifecycleevent import modified
+
+import json
+import logging
+import six
+
+try:
+    from plone.app.iterate.dexterity import ITERATE_RELATION_NAME
+    from plone.app.iterate.dexterity.relation import StagingRelationValue
+
+    # Plone 4 import error
+    from plone.app.linkintegrity.handlers import modifiedContent
+    from plone.app.linkintegrity.utils import referencedRelationship
+except ImportError:
+    ITERATE_RELATION_NAME = None
+    StagingRelationValue = None
+
+    # Plone 4 corrected paths
+    from plone.app.linkintegrity.handlers import modifiedDexterity as modifiedContent
+    from plone.app.linkintegrity.handlers import referencedRelationship
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_field_and_schema_for_fieldname(field_id, fti):
```

Comment:
DRY: this function is already in the controlpanel PR https://github.com/plone/Products.CMFPlone/pull/3232 - see also my comments there.

Reply:
The control panel is for Plone 6, and we want to have `plone.api.relation` in Plone 5. So we cannot share the code.